tests/test_content_manifest.py

```diff
@@ -206,48 +206,96 @@ def test_process_go_package_with_local_dep_unprocessed_module(default_request: R
+                    "dependencies": [],
+                },
+                {
+                    "name": "example.com/org/project",
+                    "type": "go-package",
+                    "version": "1.1.1",
+                    "dependencies": [
+                        {
+                            "name": "example.com/org/project/foo/package",
+                            "type": "go-package",
+                            "version": "./foo/package",
+                        },
+                    ],
                 },
             ],
-        },
-        {
-            "name": "example.com/org/project/foo",
-            "type": "gomod",
-            "version": "2.2.2",
-            "dependencies": [],
-        },
-        {
-            "name": "example.com/org/project",
-            "type": "go-package",
-            "version": "1.1.1",
-            "dependencies": [
+            "pkg:golang/example.com%2Forg%2Fproject@1.1.1",
+            "pkg:golang/example.com%2Forg%2Fproject%2Ffoo@2.2.2",
+            "pkg:golang/example.com%2Forg%2Fproject%2Ffoo%2Fpackage@2.2.2",
+            id="simple",
+        ),
+        pytest.param(
+            [
+                {
+                    "name": "go.etcd.io/etcd",
+                    "type": "gomod",
+                    "version": "3.5.1",
+                    "dependencies": [],
+                },
+                {
+                    "name": "go.etcd.io/etcd/client",
+                    "type": "gomod",
+                    "version": "3.5.1",
+                    "dependencies": [
+                        {
+                            "name": "go.etcd.io/etcd",
+                            "type": "gomod",
+                            "version": "../",
+                        }
+                    ],
+                },
                 {
-                    "name": "example.com/org/project/foo/package",
+                    "name": "go.etcd.io/etcd/client",
                     "type": "go-package",
-                    "version": "./foo/package",
+                    "version": "3.5.1",
+                    "dependencies": [
+                        {
+                            "name": "go.etcd.io/etcd/client/pkg",
+                            "type": "go-package",
+                            "version": "./client/pkg",
+                        },
+                    ],
                 },
             ],
-        },
-    ]
-
-    root_module_purl = "pkg:golang/example.com%2Forg%2Fproject@1.1.1"
-    foo_module_purl = "pkg:golang/example.com%2Forg%2Fproject%2Ffoo@2.2.2"
-    foo_package_purl = "pkg:golang/example.com%2Forg%2Fproject%2Ffoo%2Fpackage@2.2.2"
+            "pkg:golang/go.etcd.io%2Fetcd%2Fv3@v3.5.1-0.20230511082319-da125abcfa3f",
+            "pkg:golang/go.etcd.io%2Fetcd%2Fv3%2Fcontrib%2Flock%2Fclient@v3.5.1-0.20230511082319-da125abcfa3f",
+            "pkg:golang/go.etcd.io%2Fetcd%2Fclient%2Fpkg%2Fv3%2Fsrv@v3.5.1-0.20230511082319-da125abcfa3f",
```

Comment:
These need fixing

One comment on the `root_module_purl`, `dep_module_purl`, `dep_package_purl` setup: I think that the "root" will have to be the purl for `go.etcd.io/etcd/client`, dep will be the one for `go.etcd.io/etcd` and dep_package the one for `go.etcd.io/etcd/api` (if you go with that)

The naming is a bit off since the "root" is not the root anymore, but basically: the root is the one that has dependencies, and the two dep_ ones are its dependencies